src/main/java/spring/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,31 @@
+package spring.dao.impl;
+
+import java.util.List;
+import javax.persistence.TypedQuery;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import spring.dao.UserDao;
+import spring.model.User;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
I am not sure whether autowired is necessary here. You can remove it

Reply:
yes, thanks